libbeat/outputs/elasticsearch/client.go

```diff
@@ -61,6 +62,7 @@ func NewClient(
 	esURL, index string, proxyURL *url.URL, tls *tls.Config,
 	username, password string,
 	params map[string]string,
+	onConnectCallback *func(client *Client),
```

Comment:
we've got so many parameters in NewClient already, we might consider to pass a config struct to clean this up a little. Pretty annoying to adjust tests every single parameter we add (even if not used by tests)